packages/utils/src/store/ReactStore.ts

```diff
@@ -214,16 +235,63 @@ export class ReactStore<
   /**
    * Returns a stable setter function for a specific key in the store's state.
    * It's commonly used to pass as a ref callback to React elements.
+   *
    * @param key Key of the state to set.
    */
-  public getElementSetter<Key extends keyof State, Value extends State[Key]>(key: keyof State) {
+  public useStateSetter<Key extends keyof State, Value extends State[Key]>(key: keyof State) {
     return React.useCallback(
-      (element: Value) => {
-        this.set(key, element);
+      (value: Value) => {
+        this.set(key, value);
       },
       [key],
     );
   }
+
+  /**
+   * Observes changes derived from the store's selectors and calls the listener when the selected value changes.
+   *
+   * @param key Key of the selector to observe.
+   * @param listener Listener function called when the selector result changes.
+   */
+  public observeSelector<Key extends keyof Selectors>(
+    selector: Key,
+    listener: (
+      newValue: ReturnType<Selectors[Key]>,
+      oldValue: ReturnType<Selectors[Key]>,
+      store: this,
+    ) => void,
+  ): () => void;
+
+  public observeSelector<Selector extends ObserveSelector<State>>(
+    selector: Selector,
+    listener: (newValue: ReturnType<Selector>, oldValue: ReturnType<Selector>, store: this) => void,
+  ): () => void;
+
+  public observeSelector(
```

Comment:
If we agree to go with selectors as the only way to read the state, then we can rename this to just `.observe()`.